logrotate/src/LogRotateHelper.ipp

```diff
@@ -403,9 +427,9 @@ bool LogRotateHelper::createCompressedFile(std::string file_name, std::string gz
         return false;
     }
 
-    int N;
+    size_t N;
     while ((N = fread(buffer, 1, buffer_size, input)) > 0) {
-        gzwrite(output, buffer, N);
+        gzwrite(output, buffer, (unsigned int)N);
```

Comment:
Fixed this one, but regardless there were so many implicit castings in the .ipp file. which is what I was trying to address.